integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java

```diff
@@ -972,6 +981,59 @@ public int symlink(String linkname, String path) {
     return -ErrorCodes.ENOTSUP();
   }
 
+  /**
+   * Gets the filesystem statistics.
+   *
+   * @param path The FS path of the directory
+   * @param stbuf Statistics of a filesystem
+   * @return 0 on success, a negative value on error
+   */
+  @Override
+  public int statfs(String path, Statvfs stbuf) {
+    return AlluxioFuseUtils.call(LOG, () -> statfsInternal(path, stbuf),
+        "fuse.statfs", "path=%s", path);
+  }
+
+  private int statfsInternal(String path, Statvfs stbuf) {
+    ClientContext ctx = ClientContext.create(mConf);
+
+    try (BlockMasterClient blockClient =
+             BlockMasterClient.Factory.create(
+                 MasterClientContext.newBuilder(ctx).build())) {
+      Set<BlockMasterInfo.BlockMasterInfoField> blockMasterInfoFilter =
+          new HashSet<>(Arrays.asList(
+              BlockMasterInfo.BlockMasterInfoField.CAPACITY_BYTES,
+              BlockMasterInfo.BlockMasterInfoField.FREE_BYTES,
+              BlockMasterInfo.BlockMasterInfoField.USED_BYTES));
+      BlockMasterInfo blockMasterInfo = blockClient.getBlockMasterInfo(blockMasterInfoFilter);
+
+      // although user may set different block size for different files,
+      // small block size can result more accurate compute.
+      long blockSize = 4L * Constants.KB;
+      // fs block size
+      // The size in bytes of the minimum unit of allocation on this file system
+      stbuf.f_bsize.set(blockSize);
+      // The preferred length of I/O requests for files on this file system.
+      stbuf.f_frsize.set(blockSize);
```

Comment:
How could this size so small? Just 4K?

Reply:
will this affect the fuse.read(size)? will a bigger value resulting in fuse.read() with a larger size?

Reply:
Some app will set the buffer size accordingly. This flag is a heuristic one.